internal/store/store.go

```diff
@@ -265,6 +266,91 @@ func (store *Store) SetUnixTimeExpiry(obj *object.Obj, exUnixTimeSec int64) {
 	store.expires.Put(obj, uint64(exUnixTimeSec*1000))
 }
 
+// scanKeys performs a scan operation on the keypool, retrieving keys that match specified criteria.
+// The scan operation is controlled by a cursor and can be limited by a count, pattern, and key type.
+// The function takes a snapshot of the keypool to ensure that the scan is performed on a consistent view of the data.
+//
+// Arguments:
+// - cursor: An integer representing the current scan position within the keypool.
+// - count: The maximum number of keys to return in this scan operation.
+// - pattern: A string representing a glob pattern to filter keys (optional).
+// - keyType: A string representing the type of keys to include in the results (optional).
+//
+// Returns:
+// - newCursor: The updated cursor position after the scan, or 0 if the end is reached.
+// - keys: A slice of strings containing the keys that match the scan criteria.
+func (store *Store) ScanKeys(cursor, count int, pattern, keyType string) (newCursor int, keys []string) {
+	keys = make([]string, 0, count)
+	currentIndex := 0
+	processed := 0
+
+	store.store.All(func(k string, v *object.Obj) bool {
+		if currentIndex < cursor {
```

Comment:
So instead of All() method should I create a bucket that will keep the order same, it is a very vague idea but need to understand more about it

Reply:
Even if you do that, how do you ensure that the iteration doesn't get effected if the Map is resized?
This is a problem that involves having the map to have certain characteristics.
I am not sure of the importance/urgency of the PR.
If it is not a very important feature to be supported, my suggestion would be to park for the future.
@JyotinderSingh @arpitbbhayani wdyt?

Reply:
@soumya-codes what if we add a TODO here with the relevant details about the improvements required in the current approach? We can probably work to make this command more consistent with the redis implementation in a separate effort.

Reply:
@JyotinderSingh we can if we impose a restriction that SCAN would return all the keys in single shot. This can be our first step. Beyond this if we have to support options such as MATCH, CURSOR etc, we will need to have a more specialised implementation for both iteration as well as the hash-map itself.

Reply:
In that case it may be better to pause work on this PR for now. We should only add support for commands if we are able to implement it accurately for the most part.